From bf797c5ea8f4deff0bf067fba2c7dd2e6481aee9 Mon Sep 17 00:00:00 2001
From: Clemens Lang <cllang@redhat.com>
Date: Tue, 23 Apr 2024 10:36:33 +0200
Subject: [PATCH] build: Install provider in correct path

OpenSSL's libcrypto.pc file does expose the correct path for providers
in the modulesdir pkg-config variable (queryable using pkg-config
--variable=modulesdir libcrypto), and we do obtain that path in
./meson.build.

However, that variable is unused at the moment, and instead
src/meson.build "guesses" that path as $libdir/ossl-modules, which may
not actually be correct.

Signed-off-by: Clemens Lang <cllang@redhat.com>
Upstream-Status: Submitted [https://github.com/latchset/pkcs11-provider/pull/380]
---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index 631d474..11aacc8 100644
--- ./src/meson.build
+++ ./src/meson.build
@@ -34,5 +34,5 @@ pkcs11_provider = shared_module(
   link_depends: [pkcs11_provider_map],
   link_args: pkcs11_provider_ldflags,
   install: true,
-  install_dir: get_option('libdir') / 'ossl-modules',
+  install_dir: provider_path,
 )
-- 
2.44.0

